lib/flutter_zoom_drawer.dart

```diff
@@ -380,108 +552,160 @@ class _ZoomDrawerState extends State<ZoomDrawer>
       );
     }
+
+    // Add layer - Angle
+    // Works on Style 1 only
     if (widget.angle != 0 && widget.style != DrawerStyle.style1) {
-      final rotationAngle =
-          (((widget.angle) * pi * _rtlSlide) / 180) * _percentOpen;
-      _mainScreenContent = Transform.rotate(
-        angle: rotationAngle,
+      final _rotationAngle =
+          (((widget.angle) * pi * _rtlSlide) / 180) * _animationValue;
+      _mainScreen = Transform.rotate(
+        angle: _rotationAngle,
         alignment: widget.isRtl
             ? AlignmentDirectional.topEnd
             : AlignmentDirectional.topStart,
-        child: _mainScreenContent,
+        child: _mainScreen,
       );
     }
+
+    // Add layer - Overlay blur
     if (widget.overlayBlur != null) {
-      final blurAmount = widget.overlayBlur! * _percentOpen;
-      _mainScreenContent = ImageFiltered(
-        imageFilter: ImageFilter.blur(sigmaX: blurAmount, sigmaY: blurAmount),
-        child: _mainScreenContent,
+      final _blurAmount = widget.overlayBlur! * _animationValue;
+      _mainScreen = ImageFiltered(
+        imageFilter: ImageFilter.blur(sigmaX: _blurAmount, sigmaY: _blurAmount),
+        child: _mainScreen,
       );
     }
-    return _mainScreenContent;
+
+    // Add layer - AbsorbPointer
+    /// Prevents touches to mainScreen while drawer is open
+    if (widget.mainScreenAbsorbPointer) {
+      _mainScreen = Stack(
+        children: [
+          _mainScreen,
+          ValueListenableBuilder(
+            valueListenable: _absorbingMainScreen,
+            builder: (_, bool _valueNotifier, ___) {
+              if (_valueNotifier && stateNotifier.value == DrawerState.open) {
+                return AbsorbPointer(
+                  child: Container(
+                    color: Colors.transparent,
+                    width: MediaQuery.of(context).size.width,
+                    height: MediaQuery.of(context).size.height,
+                  ),
+                );
+              }
+              return const SizedBox.shrink();
+            },
+          ),
+        ],
+      );
+    }
+
+    return _mainScreen;
   }
 
   @override
   Widget build(BuildContext context) {
-    if (widget.disableGesture) return renderLayout();
-
     return GestureDetector(
-      /// Detecting the slide amount to close the drawer in RTL & LTR
-      onPanUpdate: (details) {
-        if (_state == DrawerState.open &&
-            ((details.delta.dx < -widget.swipeOffset && !widget.isRtl) ||
-                (details.delta.dx > widget.swipeOffset && widget.isRtl))) {
-          toggle();
-        }
-      },
-      onTap: () {
-        if (widget.mainScreenTapClose && _state == DrawerState.open) {
-          return close();
-        }
-      },
+      onTap: _onTap,
       child: renderLayout(),
     );
   }
 
   Widget renderLayout() {
-    if (widget.drawerStyleBuilder != null) return renderCustomStyle();
+    Widget _parentWidget = renderDefault();
+
     switch (widget.style) {
```

Comment:
add `if (widget.drawerStyleBuilder != null) {_parentWidget = renderCustomStyle();} else {// Add the switch statement here//}`